storage/innobase/row/row0merge.cc

```diff
@@ -1971,38 +1971,6 @@ row_merge_read_clustered_index(
 			mem_heap_empty(row_heap);
 
 			if (!mtr_started) {
-				goto scan_next;
-			}
-
-			if (clust_index->lock.is_waiting()) {
-				/* There are waiters on the clustered
-				index tree lock, likely the purge
-				thread. Store and restore the cursor
-				position, and yield so that scanning a
-				large table will not starve other
-				threads. */
```

Comment:
Looks like it was introduced by "MDEV-24142: Replace InnoDB rw_lock_t with sux_lock". Perhaps it is no longer relevant ? It is not going to have any functional effect anyway.